examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp

```diff
@@ -0,0 +1,178 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "OTAProviderExample.h"
+
+#include <app/clusters/ota-provider/ota-provider-delegate.h>
+#include <app/common/gen/cluster-id.h>
+#include <app/common/gen/command-id.h>
+#include <app/util/af.h>
+#include <protocols/secure_channel/PASESession.h> // For chip::kTestDeviceNodeId
+#include <support/RandUtils.h>
+
+#include <string.h>
+
+using namespace chip;
+
+using chip::app::clusters::OTAProviderDelegate;
+
+constexpr uint8_t kUpdateTokenLen    = 32;                      // must be between 8 and 32
+constexpr uint8_t kUpdateTokenStrLen = kUpdateTokenLen * 2 + 1; // Hex string needs 2 hex chars for every byte
+constexpr size_t kUriMaxLen          = 256;
+
+void GetUpdateTokenString(const chip::ByteSpan & token, char * buf, size_t bufSize)
+{
+    const uint8_t * tokenData = static_cast<const uint8_t *>(token.data());
+    size_t minLength          = chip::min(token.size(), bufSize);
+    for (size_t i = 0; i < (minLength / 2) - 1; ++i)
+    {
+        snprintf(&buf[i * 2], bufSize, "%02X", tokenData[i]);
+    }
+}
+
+void GenerateUpdateToken(uint8_t * buf, size_t bufSize)
+{
+    for (size_t i = 0; i < bufSize; ++i)
+    {
+        buf[i] = GetRandU8();
+    }
+}
+
+bool GenerateBdxUri(const Span<char> & fileDesignator, Span<char> outUri, size_t availableSize)
+{
+    const size_t kBdxPrefixLen          = 7;
+    const char bdxPrefix[kBdxPrefixLen] = "bdx://";
```

Comment:
does ` const char bdxPrefix[] = "bdx://"` work too not need the prefix len constant?

Reply:
yes you're correct.